_posts/2017-02-05-Retrospective-Swift-at-Artsy.md

```diff
@@ -0,0 +1,337 @@
+
+With respect to Swift on Android, potentially, logic code could be shared between platforms but realistically for our setup that's just not worth it. We're moving that kind of logic into the GraphQL instance and sharing across _all_ clients, not only native platforms. If you're sharing model code, you could generate that per-project instead from the server. Since GraphQL is strongly-typed, we're doing this for both [TypeScript + GraphQL][gql2ts] and [TypeScript + Relay][vscode-relay].
+
+We don't know where this will end, but we've prototyped porting one of our view controllers from React Native [to a website][relational-rnw]. It's almost source-compatible. This such a completely different mind space from where we were a year ago.
+
+#### Owning the stack
+
+Pick an abstraction level of our application above UIKit and we can fork it. All our tools can be also be forked. We can fix our own issues.
+
+In native, there are no concepts like, _"We'll use Steipete's fork of UIKit for UIPopover rotation fixes"_ or _"My version of Xcode will run tests when you press save."_. Well, hopefully the latter [may be fixed][xcode-extensions] in time, but the "you have no choice but to wait, and maybe it won't happen" aspect is part of the problem. You have your tools given to you, in a year you get some new ones and lose some old ones. In contrast, we've built [many][vscode-jest] [extensions][vscode-rns] [for][vscode-relay] [VS][vscode-common] [Code][vscode-danger] for our own use, and helped out on [major ones][flow-vscode]. When the VS Code didn't do what I wanted, I started using [use my own fork][essence].
+
+> Reference from JS 2017: [VS Code][code]
+
+In the last year, we have submitted code to major JavaScript dependencies of ours: React Native, Relay, VS Code, Jest and a few libraries in-between - fixing problems where we see them, offering features if we need them. Some of these changes are [small][vscode-toolbars], but some [are][relay-id] [big][jest-editor] [moves][react-shadow]. Being able to help out on any problem makes it much easier to live with the [593 dependencies](/blog/2016/08/15/React-Native-at-Artsy/) that using React Native brings.
+
+It's worth highlighting that all of this is done on GitHub, in the open. We can write issues, get responses, and have direct line to the people who are working on something we depend on. This is a stark contrast to the Radar system used internally at Apple, and which external developers have write-only access to. For external contributors radar is opaque, and [totally feels like a waste of time][tnw-radar]. On the other hand, a GitHub issue doesn't have to have to wait for the repo maintainers, others can get value from it and it's publicly indexed. If we had put all our effort into Radars instead of [issues like][eigen_launch] this, the whole community would be worse off.
```

Comment:
I mean, if you would link to _that_ video instead of the GTFO article, then I’d _totally_ be on board 😄